docs/src/pages/guides/migration-v3/migration-v3.md

```diff
@@ -0,0 +1,192 @@
+
+> This document is a work in progress.
+Have you upgraded your site and run into something that's not covered here?
+[Add your changes on GitHub](https://github.com/mui-org/material-ui/blob/next/docs/src/pages/guides/migration-v3/migration-v3.md)
+
+## Introduction
+
+This is a reference for upgrading your site from Material-UI v3 to v4.
+While there's a lot covered here, you probably won't need to do everything for your site.
+We'll do our best to keep things easy to follow, and as sequential as possible so you can quickly get rocking on v4!
+
+## Why you should migrate
+
+This documentation page covers the *how* of migrating from v3 to v4.
+The *why* is covered in the release blog post: [*Work in progress, on Medium*](https://medium.com/material-ui).
+
+## Updating Your Dependencies
+
+The very first thing you will need to do is update your dependencies
+
+### Update Material-UI version
+
+You need update your `package.json` to use the latest version of Material-UI.
+
+```json
+"dependencies": {
+  "@material-ui/core": "^4.0.0-alpha.0"
+}
+```
+
+Or run
+
+```sh
+npm install @material-ui/core@next
+
+or
+
+yarn add @material-ui/core@next
+```
+
+### Update React version
+
+The minimum required version of React was increased from `react@^16.3.0` to `react@^16.8.0`.
+This allows us to rely on [Hooks](https://reactjs.org/docs/hooks-intro.html).
+
+## Handling Breaking Changes
+
+### Styles
+
+- Remove the first option argument of `withTheme()`. The first argument was a placeholder for a potential future option. We have never found a need for it. It's time to remove this argument. It matches the emotion and styled-components API.
+
+  ```diff
+  -const DeepChild = withTheme()(DeepChildRaw);
+  +const DeepChild = withTheme(DeepChildRaw);
+  ```
+
+### Theme
+
+- The `theme.palette.augmentColor()` method no longer performs a side effect on its input color.
+In order to use it correctly, you have to use the output of this function.
+
+  ```diff
+  -const background = { main: color };
+  -theme.palette.augmentColor(background);
+  +const background = theme.palette.augmentColor({ main: color });
+
+  console.log({ background });
+  ```
+
+### Typography
+
+- Remove the deprecated typography variants. You can upgrade by performing the following replacements:
+  - display4 => h1
+  - display3 => h2
+  - display2 => h3
+  - display1 => h4
+  - headline => h5
+  - title => h6
+  - subheading => subtitle1
+  - body2 => body1
+  - body1 (default) => body2 (default)
+- Remove the opinionated `display: block` default typograpghy style.
+You can use the new `display?: 'initial' | 'inline' | 'block';` property.
+- Rename the `headlineMapping` property to better align with its purpose.
+
+  ```diff
+  -<MuiTypography headlineMapping={headlineMapping}>
+  +<MuiTypography variantMapping={variantMapping}>
+  ```
+
+### Button
+
+- Remove the deprecated button flat, raised and fab variants:
```

Comment:
The word button seems a bit redundant here, I would remove it or shuffle the sentence around.

```suggestion
- Remove the deprecated flat, raised and fab variants:
```

```suggestion
- Remove the deprecated button variants (flat, raised and fab):
```